### PR TITLE
Dump system info and processing status to file

### PR DESCRIPTION
**Pull request by godsic**
At the moment system properties and processing status is not exposed to client software. Some go as far as parsing the web gui page to figure out this information. Lets make thier life easier and dump potentially useful information to json file, which can also be accessed over web. This PR addresses #108. [`gitpython`](https://pypi.org/project/GitPython/) module is now *recommended*.

### CODE REVIEW

**Comment by godsic** [comment]
@dubrsl @alexeysirenko Please provide your comments guys.

**Comment by redradist** [comment]
@godsic LGTM

**Comment by dubrsl** [comment]
Sorry, but add dependency to SCM git in application is bad idea.
1. Someone may be using a different SCM (Source Control Management). Like mercurial, svn or smth. proprietar.
2. By default, BuildKit doesn't keep the .git directory when using Git contexts. https://docs.docker.com/build/building/context/#keep-git-directory The `.git` folder can be deleted to reduce image size or disk space usage.

I suggest simply exporting the manually set value. The value store in a static configuration file, which will be updated when a new release is released.

File: _signal_processing/krakenSDR_signal_processor.py [hunk 1]
```diff
@@ -283,6 +291,48 @@ def find_nearest(array, value):
...
+            daq_status["noise_source_enabled"] = bool(self.module_receiver.iq_header.noise_source_state)
+            daq_status["adc_overdrive"] = bool(self.module_receiver.iq_header.adc_overdrive_flags)
+            daq_status["sampling_frequency_hz"] = self.module_receiver.iq_header.adc_sampling_freq
+            daq_status["bandwidth_hz"] = self.module_receiver.iq_header.sampling_freq
+            daq_status["decimated_bandwidth_hz"] = self.module_receiver.iq_header.sampling_freq // self.dsp_decimation
+            daq_status["buffer_size_ms"] = (
+                self.module_receiver.iq_header.cpi_length / self.module_receiver.iq_header.sampling_freq
+            ) * 1e3
+            daq_status["dropped_frames"] = self.dropped_frames
+
```
Comment (dubrsl) [inline _signal_processing/krakenSDR_signal_processor.py:327]: Might make sense adding a general status. By checking which you can see whether the receiver is ready or not.

Comment (godsic) [reply _signal_processing/krakenSDR_signal_processor.py:328]: Fair enough, added `daq_ok`.

**Comment by dubrsl** [comment]
> @dubrsl Thanks for your feedback. Unfortunately, I cannot follow the first point, since the project opted for git as SCM. If you are talking about somebody who might have cloned it and turned into another SCM (e.g., Mercurial), then I don't think we should support that case. 

Ok.

> Regarding second one, I see your point for sure. Nevertheless I am not a fan of such manual steps, especially, given the fact that some of us run `krakensdr_doa` not aligned with releases and knowing the actual commit hash is beneficial for debugging purposes.

I'm sure no one installs manually. And uses scripts https://github.com/krakenrf/krakensdr_docs/tree/main/install_scripts  or Dockerfile. Therefore, I suggest setting the version and git hash in _ui/_web_interface/variables.py using this script. And remove the dependency on git.
Or expect, that the git folder may not exist.

**Comment by godsic** [comment]
@dubrsl I now handle cases of missing `gitpython` module or `.git` folder.

File: _ui/_web_interface/views/system_control_card.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def get_system_control_card_layout():
                         ],
                         className="field",
                     ),
-                    html.Div("Version 1.7.0"),
+                    html.Div(f"Version {SOFTWARE_VERSION} ({SOFTWARE_GIT_HASH})"),
```
Comment (dubrsl) [inline _ui/_web_interface/views/system_control_card.py:68]: Using built metadata with version should be used via "+"
https://semver.org/#spec-item-10

Comment (godsic) [reply _ui/_web_interface/views/system_control_card.py]: @dubrsl Fixed according to your suggestion, but with short hash instead of full one.

**Comment by dubrsl** [comment]
More thoughts. Is it possible to run a custom script and include the result it provides in the status?
In this way we can add various information on sensors (CPU temperature, battery charge when using UPS), system status (LA, RAM usage, free disk space).

**Comment by godsic** [comment]
> More thoughts. Is it possible to run a custom script(s) and include the result(s) it provides in the status? In this way we can add various information on sensors (CPU temperature, battery charge when using UPS), system status (LA, RAM usage, free disk space) etc..

Most of those metrics can be retrieved using python built-in or 3rd party modules. I am happy to add those in followup.

**Pull request merged**